compiler/src/codegen/compcore.re

```diff
@@ -1889,15 +1358,41 @@ let heap_allocate = (wasm_mod, env, num_words: int) =>
         compile_bind(wasm_mod, env, ~action=BindGet, runtime_heap_ptr),
         Expression.const(
           wasm_mod,
-          const_int32(round_to_even(num_words) * 4),
+          const_int32(round_to_even(num_words + 2) * 4),
         ),
       );
     Expression.tuple_extract(
       wasm_mod,
       Expression.tuple_make(
         wasm_mod,
         [
-          compile_bind(wasm_mod, env, ~action=BindGet, runtime_heap_ptr),
+          Expression.block(
```

Comment:
This has changed to add a fake GC header to everything allocated in runtime land so values allocated here can work in GC land. The reference count is set to 1, and shouldn't ever fall below 1.

Reply:
Yep, exactly. The problem that cropped up was GC-managed files would try to incRef/decRef runtime values, but since they didn't have GC headers, this was casually trashing runtime values 🙃 

The "fakeness" comes from the fact that calling `free()` on a runtime value wouldn't be _great_. Nothing would happen, except that a useless block would be added to the free list. (Size would be 0, so it would always be skipped.)